### PR TITLE
[gdalmdimtranslate] Use GDALArgumentParser

### PR DESCRIPTION
**Pull request by elpaso**
```
Usage: gdalmdimtranslate [--help] [--long-usage] [--help-general]
                         [-if <format>]... [-of <output_format>] [-co <NAME>=<VALUE>]...
                         [[-array <array_spec]...]
                         [-arrayoption <NAME>=<VALUE>]...
                         [[-group <group_spec>]...]
                         [-subset <subset_spec>]... [-scaleaxes <scaleaxes_spec>] [-strict] [-oo <NAME>=<VALUE>]...
                         [--quiet]
                         <src_dataset> <dst_dataset>

Converts multidimensional data between different formats, and performs subsetting.

Positional arguments:
  <src_dataset>                The source dataset name. 
  <dst_dataset>                The destination file name. 

Optional arguments:
  -h, --help                   Shows short help message and exits. 
  --long-usage                 Shows long help message and exits. 
  --help-general               Report detailed help on general options. 
  -if <format>                 Format/driver name(s) to be attempted to open the input file. [may be repeated]
  -of <output_format>          Output format. 
  -co <NAME>=<VALUE>           Creation option(s). [may be repeated]
  -array <array_spec           Select a single array instead of converting the whole dataset. [may be repeated]
  -arrayoption <NAME>=<VALUE>  Option passed to GDALGroup::GetMDArrayNames() to filter reported arrays. [may be repeated]
  -group <group_spec>          Select a single group instead of converting the whole dataset. [may be repeated]
  -subset <subset_spec>        Select a subset of the data. [may be repeated]
  -scaleaxes <scaleaxes_spec>  Applies a integral scale factor to one or several dimensions. 
  -strict                      Turn warnings into failures. 
  -oo <NAME>=<VALUE>           Open option(s) for input dataset. [may be repeated]
  -q, --quiet                  Quiet mode. No progress message is emitted on the standard output. 

For more details, consult https://gdal.org/programs/gdalmdimtranslate.html
```

### CODE REVIEW

File: apps/gdalmdimtranslate_lib.cpp [hunk 1]
```diff
@@ -55,6 +56,122 @@ struct GDALMultiDimTranslateOptions
...
+            &psOptionsForBinary->aosAllowInputDrivers);
+    }
+
+    argParser->add_output_format_argument(psOptions->osFormat);
+
+    argParser->add_creation_options_argument(psOptions->aosCreateOptions);
+
+    auto &group = argParser->add_mutually_exclusive_group();
+    group.add_argument("-array")
+        .metavar("<array_spec")
```
Comment (rouault) [inline apps/gdalmdimtranslate_lib.cpp:91]:
```suggestion
        .metavar("<array_spec>")
```

File: apps/gdalmdimtranslate_lib.cpp [hunk 2]
```diff
@@ -55,6 +56,122 @@ struct GDALMultiDimTranslateOptions
...
+    argParser->add_argument("-subset")
+        .metavar("<subset_spec>")
+        .append()
+        .store_into(psOptions->aosSubset)
+        .help(_("Select a subset of the data."));
+
+    // Note: this is mutually exclusive with "view" option in -array
+    argParser->add_argument("-scaleaxes")
+        .metavar("<scaleaxes_spec>")
+        .scan<'i', int>()
```
Comment (rouault) [inline apps/gdalmdimtranslate_lib.cpp:122]: As we support multiple values separated by comma, scan<'i', int> is likely not appropriate
```suggestion
```

File: apps/gdalmdimtranslate_bin.cpp [hunk 3]
```diff
@@ -65,57 +61,41 @@ MAIN_START(argc, argv)
...
-    GDALMultiDimTranslateOptions *psOptions =
-        GDALMultiDimTranslateOptionsNew(argv + 1, &sOptionsForBinary);
-    CSLDestroy(argv);
 
-    if (psOptions == nullptr)
+    std::unique_ptr<GDALMultiDimTranslateOptions,
+                    decltype(&GDALMultiDimTranslateOptionsFree)>
+        psOptions{GDALMultiDimTranslateOptionsNew(argv + 1, &sOptionsForBinary),
+                  GDALMultiDimTranslateOptionsFree};
+
```
Comment (rouault) [inline apps/gdalmdimtranslate_bin.cpp:87]:
```suggestion
    CSLDestroy(argv);
```

File: apps/gdalmdimtranslate_lib.cpp [hunk 4]
```diff
@@ -1802,135 +1919,58 @@ GDALMultiDimTranslate(const char *pszDest, GDALDatasetH hDstDS, int nSrcCount,
...
-        }
-
-        else if (i < argc - 1 && EQUAL(papszArgv[i], "-scaleaxes"))
-        {
-            ++i;
-            CPLStringList aosScaleFactors(
-                CSLTokenizeString2(papszArgv[i], ",", 0));
-            for (int j = 0; j < aosScaleFactors.size(); j++)
-            {
-                psOptions->aosScaleFactor.push_back(aosScaleFactors[j]);
```
Comment (rouault) [inline apps/gdalmdimtranslate_lib.cpp:1860]: this way of parsing scaleaxes should be preserved for backward compatibility

File: apps/gdalmdimtranslate_lib.cpp (hunk 2 above)
Comment (elpaso) [reply apps/gdalmdimtranslate_lib.cpp]: done

File: apps/gdalmdimtranslate_lib.cpp (hunk 4 above)
Comment (elpaso) [reply apps/gdalmdimtranslate_lib.cpp]: done

**Pull request merged**